esmvalcore/_recipe.py

```diff
@@ -505,14 +505,12 @@ def _fx_list_to_dict(fx_vars):
 
 def _update_fx_settings(settings, variable, config_user):
     """Update fx settings depending on the needed method."""
-
-    # get fx variables either from user defined attribute or fixed
-    def _get_fx_vars_from_attribute(step_settings, step_name):
-        user_fx_vars = step_settings.get('fx_variables')
-        if isinstance(user_fx_vars, list):
-            user_fx_vars = _fx_list_to_dict(user_fx_vars)
-            step_settings['fx_variables'] = user_fx_vars
-        if not user_fx_vars:
+    # Add default values to the option 'fx_variables' if it is not explicitly
+    # specified and transform fx variables to dicts
+    def _update_fx_vars_in_settings(step_settings, step_name):
```

Comment:
why change the function name? Note that now mama function and baby function have very resemblant names

Reply:
A `get_x` function should return `x`; this function only updates a `dict` in-place. I think the new name is better for that, but open for debate :grin:

Reply:
eugh, ok, am not gonna make a scene :laughing: